int8 device_id
Current[] currents
Power[] powers